app/src/common/hooks/useAccordionTabsState.js

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2021 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useState } from 'react';
+
+export const useAccordionTabsState = (initialState) => {
+  const [tabsState, setTabsState] = useState(initialState);
+  const toggleTab = (tabId) => {
+    setTabsState({ ...tabsState, [tabId]: !tabsState[tabId] });
+  };
+  const collapseTabsExceptCurr = (currentTab) => {
+    const newTabsState = Object.keys(tabsState).reduce(
+      (acc, id) => ({ ...acc, [id]: currentTab === id }),
+      {},
+    );
+    setTabsState(newTabsState);
+  };
+  return {
+    tabs: tabsState,
+    toggleTab,
+    collapseTabsExceptCurr,
+  };
```

Comment:
Please follow the pattern of array destructuring to implement custom hooks. This will make the hook more flexible.